code/modules/mining/machine_vending.dm

```diff
@@ -93,13 +93,17 @@
 	)
 	prize_list["Extra"] = list() // Used in child vendors
 
+/obj/machinery/mineral/equipment_vendor/proc/remove_id()
+	if(inserted_id)
+		inserted_id.forceMove(get_turf(src))
+		inserted_id = null
+		return TRUE
+
 /obj/machinery/mineral/equipment_vendor/power_change()
 	if(!..())
-		return
-	update_icon(UPDATE_ICON_STATE)
-	if(inserted_id && !(stat & NOPOWER))
-		visible_message("<span class='notice'>The ID slot indicator light flickers on \the [src] as it spits out a card before powering down.</span>")
-		inserted_id.forceMove(loc)
+		update_icon(UPDATE_ICON_STATE)
```

Comment:
I'm not sure about this.
If you're relying on the return value, it looks like the root `power_change()` returns FALSE if the power hasn't actually changed, meaning this might get called more often than it needs to?

Reply:
correct. you need to check that the machine has the `NOPOWER` flag before spitting out the ID, considering you just opened 20-30 PRs, I doubt this has been thoroughly tested. Otherwise you would have noticed that there is unintended behaviour (i.e. the ID being spit out when `power_change()` is called without `stat` being changed)